pkg/api/v1/ec2_router.go

```diff
@@ -50,6 +50,19 @@ func GetEc2MetadataItemPath(itemPath string) string {
 	return path.Join(V20090404URI, fmt.Sprintf("%s/%s", Ec2MetadataURI, trimmed))
 }
 
+// GetEc2MetadataItemPathWithoutTrim is used to test routing edge cases where
+// the trailing '/' is kept
+func GetEc2MetadataItemPathWithoutTrim(itemPath string) string {
```

Comment:
Does this need to be part of the apiv1 package? If this is only used for testing, this can be an unexported function within the file doing the testing.

Reply:
Good point - I've now moved that into the test file, and eliminated some duplicated code as well.